test/integration/http2_upstream_integration_test.cc

```diff
@@ -306,17 +306,24 @@ TEST_P(Http2UpstreamIntegrationTest, UpstreamConnectionCloseWithManyStreams) {
                                                             {":authority", "host"}});
     encoders.push_back(&encoder_decoder.first);
     responses.push_back(std::move(encoder_decoder.second));
+
+    // Ensure we've established a connection before issuing the resets.
+    // TODO(snowp): Why can't we handle resets here?
+    if (!fake_upstream_connection_) {
+      ASSERT_TRUE(
+          fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_));
+    }
+
     // Reset a few streams to test how reset and watermark interact.
     if (i % 15 == 0) {
       codec_client_->sendReset(*encoders[i]);
     } else {
       codec_client_->sendData(*encoders[i], 0, true);
     }
   }
-  ASSERT_TRUE(fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_));
   for (uint32_t i = 0; i < num_requests; ++i) {
-    FakeStreamPtr stream;
     upstream_requests.emplace_back();
+    FakeStreamPtr stream;
```

Comment:
This variable move isn't explicitly needed, right? Just trying to understand.

Reply:
no it's not, just a remnant of me playing around. in fact, it's not even used so i'll probably just remove it